### ISSUE TITLE
Design ETM -> CTM coupling 

### ISSUE THREAD

**Issue by noracato** [quintel/etmodel#3898]
### Couple to the CTM from the ETM
Now that the ETM is being coupled to the CTM we can start to think about how to handle this the other way around. What would it take (technically) to connect to the CTM from the ETM? 

Below you can see a few impressions and a small idea of what kind of development that would entail, and how that could look like. But this plan needs to be better worked out. Research and discussions are neccesary to make the right choices before we can start development.

### A few impressions

1. A user is coupling a CTM scenario inside the ETM
![mockup_1](https://user-images.githubusercontent.com/14875123/150952843-d47a0b82-50f7-4f31-a837-2185b323e372.png)

2. The ETM after the CTM scenario has been coupled
![mockup_2](https://user-images.githubusercontent.com/14875123/150952938-f1c0db78-4741-4964-9ac2-85fd60fc6a70.png)

### Pro's and cons

For coupling to the CTM from the ETM, I can think of the following pro's and cons.

| Pro's | Cons |
| --- | --- |
| + The ETM can send updates to the CTM | - Development work for both ETM and CTM (availability) |
| + Introduces the CTM to ETM users (bigger audience) | - Has to be very user friendly, or there is a chance that people will not find/use it|
| + Makes the current one-way coupling more clear |  |
| + Can be implemented in small steps (easy to create an MVP where we can expand on) | |

### Research needed
Below are a few questions that we should answer or research before we can start. This list is by any means not final and should be expanded on as we start working on this issue. 
 - Make choices and a design plan of overall interaction (with input from CTM)
   - What is needed from the ETM perspective?
   - What is needed from the CTM perspective? E.g. does the CTM expect the ETM to set inputs with them?
   - Create specifications of what we would need from the CTM API
   - Create specifications of API alterations on our side
 - Divide the plans up into steps of work, MVP, nice-to-haves, and create time estimates

**Time estimate for design and specifications** 3 days

**Comment by antw** [comment]
Thanks for writing this up @noracato.

I find myself confused as to which interaction we're designing. There are four possible paths for a user to take when exchanging data between the CTM and ETM:

**Taking data from the CTM and viewing the resulting ETM scenario:**

1. **CTM export → ETM**: A CTM user, working within the CTM, exports their CTM scenario to the ETM and views the results in the ETM.
2. **CTM → ETM import**: An ETM user, working within the ETM, imports their CTM scenario into the ETM then views the results in the ETM.

**Taking data from the ETM and viewing the resulting CTM scenario:**

3. **ETM export → CTM**: An ETM user, working within the ETM, exports their ETM scenario to the CTM and views the results in the CTM.
4. **ETM → CTM import**: A CTM user, working within the CTM, imports their ETM scenario into the CTM and views the results in the CTM.

I think what you're describing above is option 2, and the CTM devs are currently implementing option 1?

If so, I think we should be clear that this is still a CTM → ETM integration, although one where the user experience is mostly from within the ETM rather than the other way around.

> Pros
> \+ The ETM can send updates to the CTM

I'm not sure how this follows from being able to import a CTM scenario into the ETM. To me this sounds like option 3?

> In the background:
> * The ETM requests the ETM user_values object from the CTM for this CTM scenario. As the CTM has the ability to generate these objects already, this should be easy.

I like this. I agree that if they're willing to expose this feature through their API, asking the CTM for ETM-compatible values would save us a lot of hassle, and reduces the likelihood of the integration breaking in the future. It's better for this translation of CTM user values to ETM user values to exist in one place – if possible – rather than the CTM and the ETM both having to implement their own versions.

**Comment by marliekeverweij** [comment]
Thanks for writing this up @antw & @noracato . Good to have an overview of the possible paths. I think this is a great starting point for the discussion with our client about which path we want to support within the MVP.
